src/FixedPointNumbers.jl

```diff
@@ -42,25 +40,202 @@ export
     # Functions
     scaledual
 
+# T => BaseType
+# f => Number of Bytes reserved for fractional part
+immutable FixedPoint{T <: Integer, f} <: Real
+    i::T
+
+    # constructor for manipulating the representation;
+    # selected by passing an extra dummy argument
+    FixedPoint(i::T, _) = new(i)
+    FixedPoint(i::Integer,_) = new(i % T)
+
+    FixedPoint(x) = convert(FixedPoint{T,f}, x)
+end
+
+# basic typealiases for signed and unsigned
+typealias Fixed{T <: Signed, f} FixedPoint{T, f}
+typealias UFixed{T <: Unsigned, f} FixedPoint{T, f}
+
+# default provided typealiases
+typealias Fixed16 Fixed{Int32, 16}
+typealias UFixed8  UFixed{UInt8,8}
+typealias UFixed10 UFixed{UInt16,10}
+typealias UFixed12 UFixed{UInt16,12}
+typealias UFixed14 UFixed{UInt16,14}
+typealias UFixed16 UFixed{UInt16,16}
+
 reinterpret(x::FixedPoint) = x.i
```

Comment:
I was thinking that we maybe should rename this to `get`. `reinterpret` is quite long and one easily writes `x.i` instead